test/gtest/cpu/mdspan.cpp

```diff
@@ -167,3 +169,58 @@ TEST(CpuMpiTests, DISABLED_distributed_mdarray_transpose) {
     for (size_t c = 3; c <= cols; c++)
       test_distributed_mdarray_transpose(r, c);
 }
+
+TEST(CpuMpiTests, distribute_to_local_transpose) {
+  using T = double;
+  std::vector<T> v;
+  if (comm_rank == 0) {
+    v.resize(n);
+  }
+
+  lib::distributed_mdarray<T, dyn_2d> dsrc(rows, cols);
+  std::experimental::mdspan<T, dyn_2d> ldst(v.data(), cols, rows);
+
+  if (comm_rank == 0)
+    fill_array(dsrc, rows, cols);
+  dsrc.fence();
+
+  if (comm_rank == 0) {
+    lib::collective::transpose(0, dsrc, std::make_optional(ldst));
+  } else {
+    lib::collective::transpose(
+        0, dsrc, std::optional<std::experimental::mdspan<T, dyn_2d>>());
+  }
+
+  if (comm_rank == 0) {
+    EXPECT_EQ(dsrc(0, 0), ldst(0, 0));
+    EXPECT_EQ(dsrc(1, 0), ldst(0, 1));
+    EXPECT_EQ(dsrc(rows - 1, cols - 2), ldst(cols - 2, rows - 1));
+  }
+}
+
+TEST(CpuMpiTests, local_to_distribute_transpose) {
+  using T = double;
+  std::vector<T> v;
+  if (comm_rank == 0) {
+    v.resize(n);
+  }
+
+  std::experimental::mdspan<T, dyn_2d> lsrc(v.data(), rows, cols);
+  lib::distributed_mdarray<T, dyn_2d> ddst(cols, rows);
+
+  if (comm_rank == 0)
+    fill_array(lsrc, rows, cols);
+
+  if (comm_rank == 0) {
+    lib::collective::transpose(0, std::make_optional(lsrc), ddst);
+  } else {
+    lib::collective::transpose(
+        0, std::optional<std::experimental::mdspan<T, dyn_2d>>(), ddst);
+  }
+
+  if (comm_rank == 0) {
+    EXPECT_EQ(lsrc(0, 0), v[0]);
+    EXPECT_EQ(lsrc(1, 0), ddst(0, 1));
+    EXPECT_EQ(lsrc(rows - 1, cols - 2), ddst(cols - 2, rows - 1));
```

Comment:
Can we try to use `check_eq` method here? It checks all items in the matrix. Currently it is executed on local mdspans only but I guess it should run well on distributed too.

Reply:
I've added this change as a separate commit